README.md

```diff
@@ -1,289 +1,607 @@
+
+The collection of files within the directory **MAY** change between container launches.  The collection of files within the directory **SHOULD NOT** change during the lifetime of the container.
+
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── certificates
+│   └── private-key
+```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into the application.
+
+A Service Binding resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` must be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTranstionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A Service Binding resource **MAY** define `reason` and `message` entries to describe the last `status` transition.
+
+## Resource Type Schema
+
+```yaml
 apiVersion: service.binding/v1alpha1
 kind: ServiceBinding
 metadata:
-  name: example-service-binding
+  name:                 # string
 spec:
-  mountPathPrefix: "/var/bindings"
-  services:
-    - group: postgres.dev
-      kind: Service
-      name: global-user-db
-      version: v1beta1
-      id: postgres-global-user
-    - group: ibmcloud.ibm.com
-      version: v1alpha1
-      kind: Binding
-      name: coligo-service-binding
-      id: coligo-service-binding
-  application:
-    name: nodejs-rest-http-crud
-    group: apps
-    version: v1
-    resource: deployments
+  name:                 # string, optional, default: .metadata.name
+  kind:                 # string, optional
+  provider:             # string, optional
+
+  application:          # PodSpec-able resource ObjectReference
+    apiVersion:         # string
+    kind:               # string
+    name:               # string
+    ...
+
+  service:              # Provisioned Service-able resource ObjectReference
```

Comment:
We can postpone this discussion to after this PR merges.

Core k8s resources like ConfigMap and Secret will never conform to our duck-type, but still should be bindable. It's worth adding explicit exceptions for these kinds. If not in core, then certainly in an extension.

The Knative Addressable duck-type is defined by a url at `.status.address.url`, but also added explicit support for v1/Service.

Reply:
Is adding this complexity to the core contract a better idea than synthesizing a Provided Service that encapsulates those types?

Reply:
> If not in core, then certainly in an extension.

That was our thought in this section of the spec (before this PR):  

```
Regular k8s resources (Ingress, Route, Service, Secret, ConfigMap etc) - An annotation in the corresponding Kubernetes resources that maps the status, spec or data properties to their corresponding binding data.
```

So after this PR, the support for regular k8s Secrets / ConfigsMaps, etc, can be done via the annotations approach.  You can essentially just map all of your `.data` as the binding structure in one shot, or cherry pick parts of it.